python_modules/dagster/dagster/_core/snap/node.py

```diff
@@ -250,12 +250,15 @@ def _by_name(
     return snap.name
 
 
-def build_graph_def_snap(graph_def: GraphDefinition) -> GraphDefSnap:
+def build_graph_def_snap(graph_def: GraphDefinition, job_def: JobDefinition) -> GraphDefSnap:
```

Comment:
just in terms of understandability, I think it'd make more sense to just thread through a bool here that's like "include_all_fields" (naming could be better) rather than thread through the entire job def

Reply:
i took the spirit of this feedback if not the letter and put the bool in build_output_def_snap with the aim of improving understandability - not a hill to die on but it seemed better to me than passing a precomputed bool around